src/lir.rs

```diff
@@ -53,22 +53,22 @@ pub enum LirLogicalOperator {
     Not,
 }
 
-pub trait Translate {
+pub trait Translate<'a> {
     type Output;
-    fn translate(&self) -> Self::Output;
+    fn translate(&'a self) -> Self::Output;
 }
 
-impl Translate for Expression {
-    type Output = LirProgram;
-    fn translate(&self) -> Self::Output {
+impl<'a> Translate<'a> for Expression {
+    type Output = LirProgram<'a>;
+    fn translate(&'a self) -> Self::Output {
         let mut lir = LirProgram::new();
         lir_translate_helper(self, &mut lir);
         lir.instructions.shrink_to_fit(); // shrink the memory
```

Comment:
Just wondering if we need to shrink here, as I think `LirProgram` is just a intermedia state and will be abandoned as soon as `CirProgram` created, and shrink also consumes time here.

Reply:
nice to have. and actually the memory of LirProgram is not returned to OS immediately.